sequencing/ATACSeq/subScripts/compareBamWithGenotypes.sh

```diff
@@ -1,34 +1,49 @@
+##               ATAC-seq pipeline STEP 6.1: Verify Genotype                                            ##
+## =====================================================================================================##
+## EXECUTION: sh ./subScripts/compareBamWithGenotype.sh <sampleName>  <vcf id>                          ||
+## - execute from pipeline's main scripts directory                                                     ||
+##                                                                                                      ||
+## DESCRIPTION: This script compares genotype information of sample with existing genotype SNPs info to ||
+## verify there is no contamination.                                                                    ||
+##                                                                                                      ||
+## INPUTS:                                                                                              || 
+## $1 -> <sampleName> sample name verify genotype with their matched VCF file                           ||
+## $2 -> <vcf id> ID within the vcf file for the input sample                                           ||
+##                                                                                                      ||
+## OUTPUTS:                                                                                             ||
+## - ALIGNED_DIR/baseRecalibrate/<sampleName>_metrics.txt,/<sampleName>_baserecal.bam,                  ||
+##   /<sampleName>_baserecal.bai                                                                        ||
+## - ALIGNED_DIR/genotypeConcordance/<sampleName>.selfSM, /<sampleName>.log, /<sampleName>.depthSM      ||
+## - Samples for which a matching VCF is not found will be printed to: $META_DIR/noVCFfound.txt         ||
+##                                                                                                      ||
+## REQUIRES:                                                                                            ||
+## - sorted bam file, VCF file with SNP chip data                                                       ||
+## - Softwares: gatk, samtools, picard, verifyBamID                                                     ||
+## - Variables in config file: ALIGNED_DIR, GENOMEFASTA, KGREF,GENODIR,VERIFYBAMID                      ||
+## - Reference genome: GENOMEFASTA: genome.fa                                                           ||
+##                                                                                                      ||
+## =====================================================================================================##
+
+## ============ ##
+##    SET UP    ##
+## ============ ##
+
+cd ${ALIGNED_DIR}/
+
+## If directory does not exist, create
+mkdir -p ${ALIGNED_DIR}/baseRecalibrate
 
 sampleName=$1
 vcfid=$2
 vcfid="${vcfid%"${vcfid##*[![:space:]]}"}" 
```

Comment:
I'm struggling to tell what this is doing. I've taken a row from `sortedBDR/0_metadata/matchedVCFIDs.txt` and tried out this line here using a value from the VCFID column.

However, this doesn't seem to actually do anything to the string. 
To elaborate:
```bash
vcfid=201023670019_R06C02_EX162 # Some VCFID from the file I mentioned
vcfid="${vcfid%"${vcfid##*[![:space:]]}"}"
echo $vcfid # returns 201023670019_R06C02_EX162 (the same value)
```

Reply:
I know its not your code, just checking if this matters or not